.github/workflows/build.yml

```diff
@@ -208,7 +208,6 @@ jobs:
           GOOS: ${{ matrix.goos }}
           GOARCH: ${{ matrix.goarch }}
           GO_TAGS: "${{ env.GO_TAGS }} netcgo"
-          CGO_ENABLED: 0
```

Comment:
Injecting myself where I have no business, but if we need CGO for Darwin builds, should we not be explicit about it? Sorry if I completely missed the context...

`CGO_ENABLED: 1`

Reply:
My understanding is that CGO_ENABLED is a tristate, and unset is different from 0 or 1.  I believe unset CGO_ENABLED is the minimal change necessary to make this work.  But it's entirely possible that I'm missing something here and that the situation is simpler (or more complex) than I believe.

Reply:
When building with `CGO_ENABLED` missing from the environment, adding `netcgo` to `GO_TAGS` does not enable proper DNS resolution on macOS.

Test:

```
% GO_TAGS=netcgo make dev
% vault login -method=oidc
Error authenticating: Put "https://vault.example.internal/v1/auth/oidc/oidc/auth_url": dial tcp: lookup vault.example.internal on 172.16.108.111:53: no such host
```

`example.internal` is configured in macOS to be sent to a DNS server at `10.10.40.41` but is still using my non VPN dns server.

Next test:

```
% CGO_ENABLED=1 GO_TAGS=netcgo make dev
==> Checking that build is using go version >= 1.17.5...
==> Using go version 1.17.6...
==> Removing old directory...
==> Building...
Number of parallel builds: 11

-->    darwin/amd64: github.com/hashicorp/vault

==> Results:
total 372120
-rwxr-xr-x  1 c5309377  staff   182M Feb  2 17:04 vault
% vault login -method=oidc
Complete the login via your OIDC provider. Launching browser to:

    https://sso.example.internal/auth/realms/test/protocol/openid-connect/auth?client_id=vault&nonce=fdd8a2f553cd61bf485507d07767d5556dc53ac1&redirect_uri=http%3A%2F%2Flocalhost%3A8250%2Foidc%2Fcallback&response_type=code&scope=openid&state=d783d93ffd93dd3b385350f55610ad20a2e7b458


Waiting for OIDC authentication to complete...
Success! You are now authenticated. The token information displayed below
is already stored in the token helper. You do NOT need to run "vault login"
again. Future Vault requests will automatically use this token.

Key                  Value
---                  -----
token                s.POgDQgfP85Hld0GWC343Ws18
token_accessor       ACeIQhKo2WLiqESysljCeFdS
token_duration       24h
token_renewable      true
token_policies       ["default"]
identity_policies    []
policies             ["default"]
token_meta_role      default
% vault --version
Vault v1.10.0-dev ('2999fb8865772a40e389438a5fc21e0e98876dcd') (cgo)
```